src/h5web/vis-packs/core/heatmap/config.tsx

```diff
@@ -48,21 +43,21 @@ function initialiseStore(onRehydrated: () => void) {
         setColorMap: (colorMap: ColorMap) => set({ colorMap }),
 
         invertColorMap: false,
-        toggleColorMapInversion: () =>
-          set((state) => ({ invertColorMap: !state.invertColorMap })),
+        toggleColorMapInversion: () => {
+          set((state) => ({ invertColorMap: !state.invertColorMap }));
+        },
 
         scaleType: ScaleType.Linear,
         setScaleType: (scaleType: ScaleType) => {
-          set((state) =>
-            state.scaleType !== scaleType
-              ? { scaleType, dataDomain: undefined } // clear `dataDomain` to avoid stale state
-              : state
-          );
+          if (scaleType !== get().scaleType) {
+            set(() => ({ scaleType, dataDomain: undefined })); // clear `dataDomain` to avoid stale state
+          }
```

Comment:
TypeScript gave me an error here after I removed `extends State`. It made me realise that we were calling `set` even when the `scaleType` wasn't changing, thus triggering one unnecessary `render` (or at least one unnecessary check for changes in all the config hooks).